src/games/strategy/engine/framework/system/Memory.java

```diff
@@ -0,0 +1,119 @@
+package games.strategy.engine.framework.system;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.ClientContext;
+import games.strategy.engine.framework.GameRunner;
+
+import java.util.Properties;
+
+public class Memory {
+  public static final String TRIPLEA_MEMORY_SET = "triplea.memory.set";
+
+  private static final String TRIPLEA_MEMORY_ONLINE_ONLY = "triplea.memory.onlineOnly";
+  // what should our xmx be approximately?
+  private static final String TRIPLEA_MEMORY_XMX = "triplea.memory.Xmx";
+  private static final String TRIPLEA_MEMORY_USE_DEFAULT = "triplea.memory.useDefault";
+
+
+
+  public static void checkForMemoryXMX() {
+    final String memSetString = System.getProperty(TRIPLEA_MEMORY_SET, "false");
+    final boolean memSet = Boolean.parseBoolean(memSetString);
+    // if we have already set the memory, then return.
+    // (example: we used process runner to create a new triplea with a specific memory)
+    if (memSet) {
+      return;
+    }
+    final Properties systemIni = GameRunner.getSystemIni();
+    if (useDefaultMaxMemory(systemIni)) {
+      return;
+    }
+    if (getUseMaxMemorySettingOnlyForOnlineJoinOrHost(systemIni)) {
```

Comment:
Another one of those long method names... They are descriptive, but very hard to read